api/redhatopenshift/HcpCluster.Management/hcpCluster-models.tsp

```diff
@@ -491,3 +510,297 @@ model HcpOpenShiftClusterAdminCredential {
+  mappings: TokenClaimMappingsProfile;
+
+  /** The claim validation rules */
+  @OpenAPI.extension("x-ms-identifiers", #["type", "requiredClaim"])
+  validationRules?: TokenClaimValidationRule[];
+}
+
+/** External Auth claim mappings profile.
+ * At a minimum username or groups must be defined.
+ */
+model TokenClaimMappingsProfile {
+  /** The claim mappings username.
+   * If not specified a default value of "sub" will be used.
+   */
+  username?: UsernameClaimProfile;
+
+  /** The claim mappings groups. */
+  groups?: GroupClaimProfile;
+}
+
+/** External Auth claim profile
+ * This configures how the groups of a cluster identity should be constructed
+ * from the claims in a JWT token issued by the identity provider. When
+ * referencing a claim, if the claim is present in the JWT token, its value
+ * must be a list of groups separated by a comma (',').
+ *
+ * For example - '"example"' and '"exampleOne", "exampleTwo", "exampleThree"' are valid claim values.
+ */
+model GroupClaimProfile {
+  /** Claim name of the external profile */
+  @minLength(1)
+  @maxLength(256)
+  claim: string;
+
+  /** Prefix for the claim external profile
+   * If this is specified prefixPolicy will be set to "Prefix" by default
+   */
+  prefix?: string;
+}
+
+/** External Auth claim profile
+ * This configures how the username of a cluster identity should be constructed
+ * from the claims in a JWT token issued by the identity provider.
+ */
+model UsernameClaimProfile {
+  /** Claim name of the external profile */
+  @minLength(1)
+  @maxLength(256)
+  claim: string;
+
+  /** Prefix for the claim external profile
+   * If this is specified prefixPolicy will be set to "Prefix" by default
+   */
+  prefix?: string;
+
+  /** Prefix policy
+   * More information here: https://github.com/openshift/api/blob/f9cb766287239d10d5baae431691348286f634c1/config/v1/types_authentication.go#L633 */
+  prefixPolicy?: string;
+}
+
+/** External Auth claim validation rule */
+model TokenClaimValidationRule {
+  /** This configures the type of the validation rule.
+   * It defaults to "RequiredClaim"
+   */
+  type?: TokenValidationRuleType = TokenValidationRuleType.RequiredClaim;
+
+  /** The required claim rule to be applied. */
+  requiredClaim?: TokenRequiredClaim;
+}
+
+/** Token required claim validation rule. */
+model TokenRequiredClaim {
+  /** Claim name for the validation profile
+   * claim is a required field that configures the name of the required claim.
+   * */
+  @minLength(1)
+  claim: string;
+
+  /** Required value
+   * requiredValue is a required field that configures the value that 'claim' must
+   * have when taken from the incoming JWT claims.
+   * If the value in the JWT claims does not match, the token
+   * will be rejected for authentication.
+   */
+  @minLength(1)
+  requiredValue: string;
+}
+
+/** Representation of the different claim validation rule types that can be configured. */
+union TokenValidationRuleType {
+  string,
```

Comment:
do we want to add this? Openshift only allows 'RequiredClaim': https://github.com/openshift/api/blob/9bbe3934f75f5c636f80e70604f9b67ff0a27f86/config/v1/types_authentication.go#L731C1-L732C1

CS sets this internally to 'RequiredClaim'. If a user provides another value here, this is going to be set to 'RequiredClaim' in CS, which may be a bit confusing.